homeassistant/components/plugwise/binary_sensor.py

```diff
@@ -89,10 +125,48 @@ def _async_process_data(self):
 
         self._is_on = data[self._binary_sensor]
 
-        self._state = STATE_ON if self._is_on else STATE_OFF
         if self._binary_sensor == "dhw_state":
             self._icon = FLOW_ON_ICON if self._is_on else FLOW_OFF_ICON
         if self._binary_sensor == "slave_boiler_state":
             self._icon = FLAME_ICON if self._is_on else IDLE_ICON
 
         self.async_write_ha_state()
+
+
+class PwNotifySensor(SmileBinarySensor, BinarySensorEntity):
+    """Representation of a Plugwise Notification binary_sensor."""
+
+    def __init__(self, api, coordinator, name, dev_id, binary_sensor):
+        """Set up the Plugwise API."""
+        super().__init__(api, coordinator, name, dev_id, binary_sensor)
+
+        self._attributes = {}
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    @callback
+    def _async_process_data(self):
+        """Update the entity."""
+        notify = self._api.notifications
+
+        for severity in SEVERITIES:
+            self._attributes[f"{severity}_msg"] = []
+
+        self._is_on = False
+        self._icon = NO_NOTIFICATION_ICON
+
+        if notify:
+            self._is_on = True
+            self._icon = NOTIFICATION_ICON
+
+            for dummy, details in notify.items():
+                for msg_type, msg in details.items():
```

Comment:
How many different messages may there be?

Reply:
See https://github.com/home-assistant/core/pull/41473#discussion_r523809720 just above which also adds the keys as per https://github.com/home-assistant/core/pull/41473#discussion_r523739922

Reply:
I was wondering about the messages not the keys. Do we know how many different messages there are?

Reply:
That we know of the 'unresponsive device', 'water pressure too low', 'heating system reports an error' ,'pump has not run for 24h ...' - But we can ask Plugwise if there is a (somewhat) limited list of notifications (not sure if they'll answer and/or if it's a fixed/determined set?

Reply:
If we would know what they are, we could create one entity per message.

Reply:
We know that the Plugwise Anna and the Plugwise Adam provide these messages. I have an Adam, I think I've seen around 4 different messages since implementing this binary_sensor with persistent notification in our custom_component. I'm expecting there to be many more as the messages sometimes relate to the controlled heating/cooling device. 
But, the occurrence is low, I would see max 1-2 message per day, sometimes max 3 during one week.

I'm curious, why all the questions? What's going on in your mind wrt this particular topic? :)

Reply:
It would be better to have one entity per measurement type.

Reply:
The XML doesn't classify other than severity and a description. Reverse engineering the measurement type from the description isn't going to work (or would mean breaking often). As per givens in the reviews the binary sensor **relevant** items would now look like something like the below items. I'm not sure splitting these into `binary_sensor.{something}notification_warning` would be more beneficial for users to include/automate upon. Splitting them between 'connected device' (i.e. a plug) or 'auxiliary' (e.g. heater/boiler/whatever) would be nice but depends on string-matching.

```
binary_sensor.adam_plugwise_notification:
  is_on: True
  warning_msg: "Device ... unresponsive ..."
```
or
```
binary_sensor.adam_plugwise_notification:
  is_on: True
  error_msg: "Heating device failure"
```
or (have both set obviously from the code if both things occur), but like @bouwew indicated it will mostly be a mere 
```
binary_sensor.adam_plugwise_notification:
  is_on: False
```

Reply:
Ok.